platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -128,6 +158,8 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
             when {
                 configurationCache && startParameter.writeDependencyVerifications.isNotEmpty() -> disabledConfigurationCacheBuildModelParameters(StartParameterBuildOptions.DependencyVerificationWriteOption.LONG_OPTION)
                 configurationCache && startParameter.isExportKeys -> disabledConfigurationCacheBuildModelParameters(StartParameterBuildOptions.ExportKeysOption.LONG_OPTION)
+                // Disable configuration cache when generating a property upgrade report, since report is generated during configuration phase, and we currently don't reference it in cc cache
+                configurationCache && startParameter.isPropertyUpgradeReportEnabled -> disabledConfigurationCacheBuildModelParameters(StartParameterBuildOptions.PropertyUpgradeReportOption.LONG_OPTION)
```

Comment:
💭 The only change in this class is, that I added this condition. I had to some code out of original method, otherwise I got:
```
The function servicesForBuildTree appears to be too complex based on Cyclomatic Complexity (complexity: 25). Defined complexity threshold for methods is set to '25' [CyclomaticComplexMethod]
```